### PR TITLE
chore: Decouple metrics utils from package settings

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Reorganised metrics and base-component code to remove dependencies on package source and version. That is to move the corresponding code to the component-toolkit (as /internal/base-component and /internal/metrics) to be shared with other projects.

### How has this been tested?

Existing tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/metrics/index.ts
Comment (pan-kot) [inline src/internal/metrics/index.ts:1]: In index.ts we inject PACKAGE_VERSION and THEME to metrics utils. This allows all code in internal/metrics except for index.ts to be shared via the component-toolkit.

File: src/internal/metrics/init-awsui-versions.ts
Comment (just-boris) [inline src/internal/metrics/init-awsui-versions.ts:16]: You probably want to make `components` part configurable too. To allow `{components: [], boards: []}` structure

File: src/internal/metrics/index.ts
Comment (just-boris) [inline src/internal/metrics/index.ts:18]: This condition is a trait of all components in our system, including new the package. 

Actually, I would extract the original `useBaseComponent` hook into the shared utils, because it contains all features we need for all components. You may keep the hooks separately in the implementation, but the utility API should expose `useBaseComponent`

File: src/internal/base-component/index.ts
Comment (just-boris) [inline src/internal/base-component/index.ts:6]: Just in case, this object is not really used in any metrics, so putting it into that folder is misleading

File: src/internal/metrics/metrics.ts
Comment (just-boris) [inline src/internal/metrics/metrics.ts:90]: This needs to be parametrised, "components" or "boards"

File: src/__tests__/use-telemetry-usage.test.tsx
Comment (just-boris) [inline src/__tests__/use-telemetry-usage.test.tsx:27]: Since it is now a separate module, it can be imported directly, no need to transfer it via test helper.

Metrics test helper need only maintain `resetOneTimeMetricsCache`, because it an internal state. Everything else are pure functions, can be imported/exported directly

File: src/internal/metrics/init-awsui-versions.ts
Comment (pan-kot) [reply src/internal/metrics/init-awsui-versions.ts:16]: Oh, I kind of thought we'd like to keep components for both. But makes sense as the package versions can be different - will update 👍

File: src/internal/metrics/index.ts
Comment (pan-kot) [reply src/internal/metrics/index.ts:18]: Ok, the only difference between `useComponentMetrics` and `useBaseComponent` is the theme and package version injection. The package injection will still happen from outside the toolkit but the theme can be handled of course. The downside is that we'd have to move or copy at least the `useVisualRefresh` hook to the toolkit to support that.

File: src/internal/base-component/index.ts
Comment (pan-kot) [reply src/internal/base-component/index.ts:6]: Makes sense, I'll create a separate folder for that.

File: src/__tests__/use-telemetry-usage.test.tsx
Comment (pan-kot) [reply src/__tests__/use-telemetry-usage.test.tsx:27]: Makes sense

File: src/alert/index.tsx
Comment (just-boris) [inline src/alert/index.tsx:7]: Any chances of not changing this in the current PR? The import path will change to `@cloudscape-design/component-toolkit` soon, so let's keep the current ones for now

File: src/internal/base-component/telemetry.ts
Comment (just-boris) [inline src/internal/base-component/telemetry.ts:7]: Why is this injected? In all use-cases, we use the same metrics provider, never changes dynamicaly

File: src/internal/base-component/index.ts
Comment (just-boris) [inline src/internal/base-component/index.ts:7]: I have lost tracking what is happening with all file moves. Let's get back to the task

1. `useBaseComponent` changes its location from this package to `@cloudscape-design/component-toolkit/internal`
2. Because it has dependency on `useVisualRefresh` hook, that hook moves to `@cloudscape-design/component-toolkit/internal` too.

After the move, we replace all imports in this in this package with their equivalents from `@cloudscape-design/component-toolkit/internal`.

Why is it needed to move that many files around? I am afraid that if we merge it the current way, the very next PR would require to undo some of the changes

Comment (pan-kot) [reply src/internal/base-component/index.ts:7]: Thanks for the feedback - I will revert some changes to reduce the amount of diff and annotate the code that can be moved to the toolkit.

File: src/internal/base-component/telemetry.ts
Comment (pan-kot) [reply src/internal/base-component/telemetry.ts:7]: The metrics object is instantiated with package source and package version. If the metrics will be instantiated here the function will no longer be suitable to move to the toolkit.

File: src/alert/index.tsx
Comment (pan-kot) [reply src/alert/index.tsx:7]: Yes, will update.

File: src/internal/base-component/telemetry.ts
Comment (just-boris) [reply src/internal/base-component/telemetry.ts:7]: Discussed in DM, proposed solution is to provide all variables via arguments

```
function useTelemetry(package: string, version: string, componentName: string, theme: string)
```

File: src/internal/utils/__tests__/metrics.test.ts
Comment (just-boris) [inline src/internal/utils/__tests__/metrics.test.ts:5]: Removing module mocks is always good. 👍

**Review by just-boris — APPROVED** [review]
Approved, but I would wait for @michaeldowseza approval too, in case if there are conflicts with his work in progress

**Pull request merged**
